### PR TITLE
test: Evals test coverage improvements

### PR DESCRIPTION
**Pull request by Patryk-Kuchta**
# Overview
Expanded test coverage for the `evals` folder. Added comprehensive integration and unit tests for transcription evaluation, AMI dataset handling, metrics computation, and summarisation utilities.

## JIRA Ticket
[AIILG-328](https://mhclgdigital.atlassian.net/browse/AIILG-328)

## Changes

- Added tests for summarisation
- Refactored and consolidated existing test files to improve maintainability
- Added integration tests for dataset loading, segment selection, and metrics aggregation
- Improved test organization by consolidating duplicate tests and removing redundant files

## Acceptance Criteria (AC)

- [x] AC1: Achieve at least 80% test coverage for the evals folder
- [x] AC2: All tests pass with no failures
- [x] AC3: All pre-commit hooks pass (ruff, ruff-format, mypy)
- [x] AC4: Tests validate real behavior with integration-style testing where appropriate

---

## Testing

- **Automated**: run the tests suite

### CODE REVIEW

File: tests/evals/transcription/test_config_validation.py
Comment (Patryk-Kuchta) [inline tests/evals/transcription/test_config_validation.py:1]: These tests were mainly testing the built-in `isinstance()` rather than anything useful

**Comment by Patryk-Kuchta** [comment]
```
evals/summarisation/src/adapter_factory.py                           18      0   100%
evals/summarisation/src/config.py                                    34      0   100%
evals/summarisation/src/dspy_wrapper.py                              28      0   100%
evals/summarisation/src/jsonl.py                                     12      0   100%
evals/summarisation/src/langchain_adapter.py                         37      0   100%
evals/summarisation/src/metric.py                                    29     12    59%   29-39, 46-61
evals/summarisation/src/prompts.py                                    7      0   100%
evals/summarisation/src/runner.py                                   122     38    69%   80-82, 100-118, 127-130, 134-139, 179-188, 193-226, 232
evals/summarisation/src/schemas.py                                   28      0   100%
evals/summarisation/src/signatures.py                                12      0   100%
evals/transcription/__init__.py                                       0      0   100%
evals/transcription/src/__init__.py                                   0      0   100%
evals/transcription/src/adapters/__init__.py                          0      0   100%
evals/transcription/src/adapters/base.py                             30      0   100%
evals/transcription/src/adapters/registry.py                          4      0   100%
evals/transcription/src/config_validation.py                         10      6    40%   18-25
evals/transcription/src/constants.py                                  6      0   100%
evals/transcription/src/core/__init__.py                              0      0   100%
evals/transcription/src/core/ami/__init__.py                          4      0   100%
evals/transcription/src/core/ami/audio.py                            29      3    90%   19, 38-39
evals/transcription/src/core/ami/cache.py                            33      0   100%
evals/transcription/src/core/ami/constants.py                         1      0   100%
evals/transcription/src/core/ami/loader.py                          130     25    81%   31-37, 131-140, 157-159, 181, 198-202, 266-268, 303
evals/transcription/src/core/ami/metadata.py                         39     30    23%   23-67
evals/transcription/src/core/ami/selection.py                        19     15    21%   18-42
evals/transcription/src/core/ami_dataset.py                          34      0   100%
evals/transcription/src/core/dataset.py                              31      7    77%   23-32
evals/transcription/src/core/metrics/__init__.py                      6      0   100%
evals/transcription/src/core/metrics/aggregation.py                  12      0   100%
evals/transcription/src/core/metrics/diarization.py                  68      1    99%   60
evals/transcription/src/core/metrics/speaker_count.py                 9      0   100%
evals/transcription/src/core/metrics/transcription.py                 6      0   100%
evals/transcription/src/core/metrics/transforms.py                    9      1    89%   20
evals/transcription/src/core/results.py                              18      0   100%
evals/transcription/src/core/runner.py                               68      0   100%
evals/transcription/src/core/segments.py                              9      1    89%   23
evals/transcription/src/evaluate.py                                  81     40    51%   53-55, 58-59, 100-102, 109-164
```

## Untested Evals Code - Why It's Not Tested

### **Summarisation Evals**

**[evaluate.py](cci:7://file:///Users/patkuc/MHCLG/i-ai-minute/evals/summarisation/src/evaluate.py:0:0-0:0)**
- **Why**: CLI entrypoint that only orchestrates already-tested functions. No additional business logic to validate.

**[runner.py](cci:7://file:///Users/patkuc/MHCLG/i-ai-minute/evals/summarisation/src/runner.py:0:0-0:0) (partial)**
- **Why**: Makes live LLM API calls requiring Azure credentials and external services. Mocking defeats the purpose of validating actual LLM behavior.

**[metric.py](cci:7://file:///Users/patkuc/MHCLG/i-ai-minute/evals/summarisation/src/metric.py:0:0-0:0) (partial)**
- **Why**: Judge evaluation requires live LLM calls to score summaries. Cannot validate judge quality without real model responses.

### **Transcription Evals**

**[evaluate.py](cci:7://file:///Users/patkuc/MHCLG/i-ai-minute/evals/summarisation/src/evaluate.py:0:0-0:0) (partial)**
- **Why**: CLI main() function is integration-level orchestration. Core logic already tested in isolation.

**AMI Dataset Utilities**
- **Why**: Dataset-specific file I/O and XML parsing with minimal business logic. Better validated through end-to-end dataset loading tests.

**[config_validation.py](cci:7://file:///Users/patkuc/MHCLG/i-ai-minute/evals/transcription/src/config_validation.py:0:0-0:0) (partial)**
- **Why**: Error message paths for type validation. Low value - error messages don't affect behavior.

**[core/ami/audio.py](cci:7://file:///Users/patkuc/MHCLG/i-ai-minute/evals/transcription/src/core/ami/audio.py:0:0-0:0) (partial)**
- **Why**: ffmpeg error handling edge cases. Requires filesystem mocking with minimal business logic impact.

**Minor gaps (metrics, transforms, segments)**
- **Why**: Edge cases like division by zero, type conversions. Defensive code paths with negligible real-world impact.

File: tests/evals/transcription/test_runner.py [hunk 1]
```diff
@@ -186,3 +186,252 @@ def test_compute_all_metrics(ref_text, hyp_text, ref_diar, hyp_diar, expected):
...
+        duration_fn=lambda _path: 2.0,
+        run_id="test_run",
+        timestamp="20240101_120000",
+        dataset_version="FakeDataset_v0",
+        dataset_split="test",
+        max_workers=2,
+    )
+
+    assert len(results) == 1
+    assert results[0].summary.engine_version == "Engine_A"
```
Comment (harrybest-softwire) [inline tests/evals/transcription/test_runner.py:263]: which of these assertions is checking that at most 2 workers were used? I'm struggling to understand this test a little.

File: tests/evals/transcription/test_runner.py [hunk 2]
```diff
@@ -186,3 +186,252 @@ def test_compute_all_metrics(ref_text, hyp_text, ref_diar, hyp_diar, expected):
...
+        wav_write_fn=lambda ex, _idx: ex.audio.path,
+        duration_fn=lambda _path: 2.0,
+        run_id="test_run",
+        timestamp="20240101_120000",
+        dataset_version="FakeDataset_v0",
+        dataset_split="test",
+    )
+
+    summary = results[0].summary
+    assert summary.n_examples == 2
```
Comment (harrybest-softwire) [inline tests/evals/transcription/test_runner.py:331]: sorry I think I'm struggling with these since I don't have the domain knowledge yet. But why does `n_examples == 2` mean that the timing as been accumulated. Is it related to the `duration_fn`? I can't see where any sort of timing thing is being invoked.

File: tests/evals/transcription/test_runner.py [hunk 3]
```diff
@@ -186,3 +186,252 @@ def test_compute_all_metrics(ref_text, hyp_text, ref_diar, hyp_diar, expected):
...
+        wav_write_fn=lambda ex, _idx: ex.audio.path,
+        duration_fn=lambda _path: 2.0,
+        run_id="test_run",
+        timestamp="20240101_120000",
+        dataset_version="FakeDataset_v0",
+        dataset_split="test",
+    )
+
+    sample = results[0].samples[0]
+    assert sample.latency_ms == pytest.approx(1500.0)
```
Comment (harrybest-softwire) [inline tests/evals/transcription/test_runner.py]: info: is this `1500.0` figure coming from `proc_sec = 1.5`?

File: tests/evals/summarisation/test_signatures.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
```
Comment (harrybest-softwire) [inline tests/evals/summarisation/test_signatures.py]: Are these to guard against breaking changes in a library. Just a little confused why me need to check that our types have specific fields?

**Review by harrybest-softwire — APPROVED** [review]
Looks good to me, to be honest I didn't understand a lot of the tests here since they're very domainy. But it looks like you have a comprehensive suite. In future I can imagine that we might want to drop some of these tests / loosen them, since they're so thorough, are there any that you think might be particularly brittle?

File: tests/evals/summarisation/test_signatures.py (hunk 4 above)
Comment (Patryk-Kuchta) [reply tests/evals/summarisation/test_signatures.py]: Yeah basically, the contract bit in the name of each test got a bit lost (refactored), but that's the key reason why we add them. It makes spotting breaking changes a ton easier and given how quick generating tests is nowadays I think its worth including :)

File: tests/evals/transcription/test_runner.py (hunk 3 above)
Comment (Patryk-Kuchta) [reply tests/evals/transcription/test_runner.py]: Yes, fair question, I made a variable to make this connection a bit clearer.

File: tests/evals/transcription/test_runner.py (hunk 1 above)
Comment (Patryk-Kuchta) [reply tests/evals/transcription/test_runner.py:263]: Agree, the assertion here got a bit lost fixed so that it does what it says on the tin :)

File: tests/evals/transcription/test_runner.py (hunk 2 above)
Comment (Patryk-Kuchta) [reply tests/evals/transcription/test_runner.py:331]: It's not your confusion, it's my mistake 😅. I improved the assertion here so that it properly tests processing duration
